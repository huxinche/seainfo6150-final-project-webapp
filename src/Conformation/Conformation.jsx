import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from "prop-types";
const Conformation = (props) => {
    const { form } = props;
    return <div className="container mt">
        <h1>Your profile</h1>
        <table>
            <tbody>
                <tr>
                    <th>Name</th>
                    <td>{form.name}</td>
                </tr>
                <tr>
                    <th>Gender</th>
                    <td>{form.gender}</td>
                </tr>
                <tr>
                    <th>Datebirth</th>
                    <td>{form.dob}</td>
                </tr>
                <tr>
                    <th>Email</th>
                    <td>{form.email}</td>
                </tr>
                <tr>
                    <th>Phone</th>
                    <td>{form.phone}</td>
                </tr>

                <tr>
                    <th>Religion</th>
                    <td>{form.religion}</td>
                </tr>
                <tr>
                    <th>Education level</th>
                    <td>{form.edu}</td>
                </tr>
                <tr>
                    <th>Your Password</th>
                    <td>{form.password}</td>
                </tr>
            </tbody>

        </table>
        <br />
        <br />
        <div style={{ textAlign: 'center' }}>
            <Link to="/" className="btn-back">Back to Home</Link>
        </div>
        <br />
        <br />

    </div>
}

Conformation.propTypes = {
    form: PropTypes.object.isRequired,
}

export default Conformation;