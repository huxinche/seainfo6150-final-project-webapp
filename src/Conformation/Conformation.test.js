import React from "react";
import Conformation from "./Conformation";

it("renders the Conformation component correctly", () => {

    const { container } = render(<Conformation form={{
        name: '',
        dob: '',
        phone: '',
        email: '',
        password: '',
        religion: null,
        edu: null,
        gender: null,
    }} />);
    expect(container).toMatchSnapshot();
})