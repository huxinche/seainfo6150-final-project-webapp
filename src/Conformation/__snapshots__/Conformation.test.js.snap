// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Conformation component correctly 1`] = `
<div>
  <div
    class="container mt"
  >
    <h1>
      Your profile
    </h1>
    <table>
      <tbody>
        <tr>
          <th>
            Name
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Gender
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Datebirth
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Email
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Phone
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Religion
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Education level
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Your Password
          </th>
          <td />
        </tr>
      </tbody>
    </table>
    <br />
    <br />
    <div
      style="text-align: center;"
    >
      <a
        class="btn-back"
        href="/"
      >
        Back to Home
      </a>
    </div>
    <br />
    <br />
  </div>
</div>
`;
